### PR TITLE
Add core.auth_oauth_login_after and core.auth_oauth_link_after

### PR DESCRIPTION
**Pull request by senky**
Checklist:
- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-14825

PHPBB3-14825

### CODE REVIEW

**Comment by marc1706** [comment]
Build errors are caused by this:
`ERROR: An error occurred while parsing "phpbb\auth\provider\oauth\oauth::__construct" line "138": "13" @param tags are expected but only "1" found`

**Comment by senky** [comment]
@marc1706 could you give me a hand with fixing this issue? I don't even understand the error message. There ARE 13 `@param` tags in the constructor's docblock...

File: phpBB/phpbb/auth/provider/oauth/oauth.php [hunk 1]
```diff
@@ -126,8 +133,9 @@ class oauth extends \phpbb\auth\provider\base
 	* @param	\Symfony\Component\DependencyInjection\ContainerInterface $phpbb_container DI container
 	* @param	string			$phpbb_root_path
 	* @param	string			$php_ext
+	* @param 	\phpbb\event\dispatcher_interface	$dispatcher phpBB Event dispatcher object
```
Comment (marc1706) [inline phpBB/phpbb/auth/provider/oauth/oauth.php]: Hard to track down but there is actually a space after `@param which the sami parse check does not like. If you remove it, the tests will run fine.

**Comment by marc1706** [comment]
Sorry, I overlooked that the oauth class already gets passed the container. You don't need to pass the dispatch via the constructor but can use `$phpbb_container->get('dispatcher');`.

File: tests/functions/user_delete_test.php [hunk 2]
```diff
@@ -59,6 +59,7 @@ protected function setUp()
 		$passwords_helper = new \phpbb\passwords\helper;
 		// Set up passwords manager
 		$passwords_manager = new \phpbb\passwords\manager($config, $passwords_drivers, $passwords_helper, array_keys($passwords_drivers));
+		$dispatcher = new phpbb_mock_event_dispatcher();
```
Comment (marc1706) [inline tests/functions/user_delete_test.php]: The dispatcher is already defined as `$phpbb_dispatcher` a few lines above this. As mentioned in my comment, you don't actually need to pass the dispatcher to the class via the constructor though.

**Comment by senky** [comment]
> Sorry, I overlooked that the oauth class already gets passed the container. You don't need to pass the dispatch via the constructor but can use $phpbb_container->get('dispatcher');.

I don't really understand why are not other services initialized by container as well, but OK.

**Comment by senky** [comment]
As build fails revealed:
https://travis-ci.org/phpbb/phpbb/builds/175502165
I really have to use DI to inject dispatcher.

**Comment by marc1706** [comment]
No, you just have to set the dispatcher service in the tests (which is what this is complaining about):
https://github.com/senky/phpbb/blob/df0388ccc5b24cfb67078dd2fa8cb2fc4f607dac/tests/functions/user_delete_test.php#L83

However, you can leave it like that now. Really not worth the trouble to change this all over again. ;)

**Pull request merged**